pkg/menu/menu.go

```diff
@@ -0,0 +1,75 @@
+package menu
+
+import (
+	"os"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+const menuWidth = 50
+
+// newParagraph will return a paragraph object with given initial text.
+func newParagraph(initText string, border bool, location int, wid int, ht int) *widgets.Paragraph {
+	p := widgets.NewParagraph()
+	p.Text = initText
+	p.Border = border
+	p.SetRect(0, location, wid, location+ht)
+	p.TextStyle.Fg = ui.ColorWhite
+	ui.Render(p)
+	return p
+}
+
+// GetInput will present an input box to user and return the user's input.
+// GetInout will check validation of input using isValid function.
+func GetInput(introwords string, location int, wid int, ht int, isValid func(string) (bool, string), testMode bool, testInput chan ui.Event) (string, error) {
```

Comment:
Suggestion: what if you had your functions set up like this

```
func GetInput(introwords string, location int, wid int, ht int, isValid func(string) (bool, string)) (string, error) {
   // show intro text, set up input and warning paragraphs
   return processInput(input, warning, isValid, ui.PollEvents())
}

func processInput(input, warning *widgets.Paragraph, isValid func(string) (bool, string), ch chan ui.Event) (string, error) {
   // have the for loop that reads from ch 
}
```
Then in your test, you can set up the channel using ui.PollEvents() and pass that channel to processInput, and then send ui events on it. Would that work?

Reply:
I think this might also allow you to test that the warnings are being correctly rendered, because if you pass the `warning` *widgets.Paragraph and pass an incorrect input, you can check that the warning text matches what you expect right?